scanpy/tests/test_get.py

```diff
@@ -51,18 +107,94 @@ def test_obs_df(adata):
     assert all(badkey_err.match(k) for k in badkeys)
 
 
+def test_backed_vs_memory():
+    "compares backed vs. memory"
+    from pathlib import Path
+
+    # get location test h5ad file in datasets
+    HERE = Path(sc.__file__).parent
+    adata_file = HERE / "datasets/10x_pbmc68k_reduced.h5ad"
+    adata_backed = sc.read(adata_file, backed='r')
+    adata = sc.read_h5ad(adata_file,)
+
+    # use non-sequential list of genes
+    genes = list(adata.var_names[20::-2])
+    obs_names = ['bulk_labels', 'n_genes']
+    pd.testing.assert_frame_equal(
+        sc.get.obs_df(adata, keys=genes + obs_names),
+        sc.get.obs_df(adata_backed, keys=genes + obs_names),
+    )
+
+    # use non-sequential list of cell indices
+    cell_indices = list(adata.obs_names[30::-2])
+    pd.testing.assert_frame_equal(
+        sc.get.var_df(adata, keys=cell_indices + ["highly_variable"]),
+        sc.get.var_df(adata_backed, keys=cell_indices + ["highly_variable"]),
+    )
+
+
+def test_column_content():
+    "uses a larger dataset to test column order and content"
+    adata = sc.datasets.pbmc68k_reduced()
+
+    # test that columns content is correct for obs_df
+    query = ['CST3', 'NKG7', 'GNLY', 'louvain', 'n_counts', 'n_genes']
+    df = sc.get.obs_df(adata, query)
+    for col in query:
+        assert col in df
+        np.testing.assert_array_equal(query, df.columns)
```

Comment:
This shouldn't happen once for every column

Reply:
True. will put it outside.